src/main/java/org/jboss/ejb/client/EJBInvocationHandler.java

```diff
@@ -277,12 +279,11 @@ private static void sendRequestWithPossibleRetries(final EJBClientInvocationCont
             }
             // retry the request
             final String failedNodeName = rsfe.getFailedNodeName();
-            if (failedNodeName != null) {
+            if (failedNodeName != null && !failedNodes.contains(failedNodeName)) {
+                failedNodes.add(failedNodeName);
                 Logs.MAIN.debugf(rsfe, "Retrying invocation %s which failed on node: %s due to:", clientInvocationContext, failedNodeName);
-                // exclude this failed node, during the retry
-                clientInvocationContext.markNodeAsExcluded(failedNodeName);
```

Comment:
What customer complained was that "Further EJB invocations are rejected with EJBCLIENT0025". Don't exclude the node in the retry process will prevent that.